cmd/snap-confine/snap-confine.c

```diff
@@ -438,6 +475,36 @@ int main(int argc, char **argv)
 		if (real_uid != 0 && (getgid() == 0 || getegid() == 0))
 			die("permanently dropping privs did not work");
 	}
+
+	// Now that we've permanently dropped, regain SYS_ADMIN
+	if (keep_sys_admin) {
+		debug("regaining SYS_ADMIN");
+		data[0].effective = CAP_TO_MASK(CAP_SYS_ADMIN);
+		data[0].permitted = data[0].effective;
+		if (capset(&hdr, data) != 0) {
+			die("capset regain failed");
+		}
+	}
+
+	// Now that we've dropped and regained SYS_ADMIN, we can load the
+	// seccomp profiles.
+	if (sc_apply_seccomp_profile_for_security_tag(invocation.security_tag)) {
+		/* If the process is not explicitly unconfined then load the global
+		 * profile as well. */
+		sc_apply_global_seccomp_profile();
+	}
+
+	// Even though we set inheritable to 0, let's clear SYS_ADMIN
+	// explicitly
+	if (keep_sys_admin) {
+		debug("clearing SYS_ADMIN");
+		data[0].effective = 0;
+		data[0].permitted = data[0].effective;
+		if (capset(&hdr, data) != 0) {
```

Comment:
```suggestion
		if (capset(&hdr, data) < 0) {
```

Reply:
This falls into the category of https://github.com/snapcore/snapd/pull/7026#discussion_r297842904 where I find the '!= 0' clearer here (since 0 is the only measure of success).